### DISCUSSION TITLE
How to use .NET MAUI Template?

### DISCUSSION THREAD

**Discussion by jackdaus** [StereoKit/StereoKit#475] [Q&A] [answered]
Hey! I am starting a new StereoKit project, and I want to try to use the (in-development) MAUI template. However, I'm running into a runtime error. 

Here are the steps I've tried so far:
1. Cloned this repo
2. Checked out the latest commit on the feature/maui_template branch (https://github.com/StereoKit/StereoKit/commit/f32fbf80d0773d1b416d33274ee2efdfb6a907b1)
3. Open the SKTemplate_Maui solution in Visual Studio 2022
4. Click Run

But shortly after I click run, I get the following error
`System.DllNotFoundException: 'Unable to load DLL 'StereoKitC' or one of its dependencies: The specified module could not be found. (0x8007007E)'`

![image](https://user-images.githubusercontent.com/53974525/190273397-0ff863d7-6eac-434e-912c-0944e7ca2c62.png)

Some notes about the environment
- Windows 11
- Fallback Simulator
- .NET framework used is net6.0-windows10.0.19041.0
- StereoKit 0.3.6-preview.6

Thanks!

**Comment by maluoi** [comment, marked as answer]
Oh, this may actually require a bit more surgery to get working! The template is somewhat dependent on features in the NuGet package that aren't present in the current preview builds.

In particular, any [changes to the target files here](https://github.com/StereoKit/StereoKit/compare/feature/maui_template), and possibly a bit of folder manipulation as well. I'm not certain I'd really recommend going down this path just yet unless you're _really_ enthusiastic about it. I am trying to put together a preview build though, so I'll look and see if I can pull in some of the changes that'll help this work!

**Reply by maluoi** [reply]
FYI, v0.3.7-preview.4 is out, and will likely behave quite a bit better with respect to this! I also merged the MAUI template into develop for the moment. It does look like there may still be some outstanding issues with compiling shaders though (or possibly asset copying?) :)

**Reply by jackdaus** [reply]
Wow awesome thanks! I'll give it another shot sometime tomorrow probably!!

**Reply by jackdaus** [reply]
Cool, I got the MAUI template working on my end with v0.3.7-preview.4! At least for the net6.0 framework. (The net6.0-android framework didn't seem to work for my Quest 2. But I don't really care at the moment since I can still run the project on my Quest 2 using Quest Link with net6.0.) And thanks for the tip about copying the shader file.

On a related note, the main reasons I want to use the MAUI template is (1) so I can use newer .NET features (instead of being tied to .NET Standard 2.0) and (2) to leave open the possibility of distributing my application to Quest in the future. So far, I have been assuming that the only way to distribute an SK app to a Quest is to build an Android application (either with Xamarin (i.e., XPlat template) or with the MAUI template). Is that correct? Or am I missing something, and it is possible to distribute to Quest even if I use the StereoKit .Net Core template? Just wanted to make sure I've got the right idea here. Thanks!
